quantestpy/test_circuit.py

```diff
@@ -524,8 +525,98 @@ def _create_all_qubit_gate_from_crz_gate(
 
         return all_qubit_gate
 
-    def set_initial_state_vector(self, initial_state_vector: np.ndarray) \
-            -> None:
+    def _create_all_qubit_gate_from_cu1_gate(
+            self,
+            control_qubit: list,
+            target_qubit: list,
+            control_value: list,
+            parameter: list) -> np.ndarray:
+
+        all_qubit_gate = np.zeros((2**self._num_qubit, 2**self._num_qubit))
+        patterns = list(itertools.product([0, 1], repeat=len(control_qubit)))
```

Comment:
@mitz1012 
Implementation is great! But it would be nicer to have a better nomenclature for `patterns` in order to show "patterns of what".